ubiquity/dell-bootstrap.py

```diff
@@ -429,43 +429,39 @@ def remove_extra_partitions(self):
             self.log(str(err))
         
         if self.test_swap():
-            swap_part = ''        
+            swap_part = ''  
        #remove extras
         for number in (os_part,swap_part):
             if number.isdigit():
                 remove = misc.execute_root('parted', '-s', self.device, 'rm', number)
                 if remove is False:
                     self.log("Error removing partition number: %s on %s (this may be normal)'" % (number, self.device))
-                #refresh = misc.execute_root('partx', '-d', '--nr', number, self.device)
-                #if refresh is False:
-                    #self.log("Error updating partition %s for kernel device %s (this may be normal)'" % (number, self.device))
+                refresh = misc.execute_root('partx', '-d', '--nr', number, self.device)
+                if refresh is False:
+                    self.log("Error updating partition %s for kernel device %s (this may be normal)'" % (number, self.device))
        
     def dual_partition_num(self,label):    
        #remove UBUNTU patition for dual boot
        ##OS num
+        os_part = ''
+        swap_part = ''        
         digits = re.compile('\d+')
         try:
             os_path = magic.fetch_output(['readlink','/dev/disk/by-label/'+label]).split('\n')            
         except Exception as err:
-            self.log('os_path command is executed failed, the error is %s'%str(err))
+            # compatible with DUALSYS partition label when boot from hdd 
+            os_path = magic.fetch_output(['readlink','/dev/disk/by-label/UBUNTU']).split('\n')
+            if not os_path:
+                self.log('os_path command is executed failed, the error is %s'%str(err))
```

Comment:
why do you need this fallback path?  Maybe it would be better to send a list of compatible labels (DUALSYS, UBUNTU etc) and test them each.